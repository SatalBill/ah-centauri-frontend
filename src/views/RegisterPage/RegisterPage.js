import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Header,
  Icon,
  Message,
} from 'semantic-ui-react';

import './RegisterPage.scss';
import { signUpAction } from '../../actions/authActions';
import CustomForm from '../../components/CustomForm/CustomForm';
import isEmpty from '../../utils/is_empty';
import GridContainer from '../../components/CustomGrid/GridContainer';
import GridItem from '../../components/CustomGrid/GridItem';


export class RegisterPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      errorMessage: {},
    };
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (prevState.errorMessage !== nextProps.errorMessage

      || prevState.loading !== nextProps.loading) {
      return {
        errorMessage: nextProps.errorMessage,
      };
    }

    return null;
  }

  handleInputChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();

    this.props.signUpAction({ user: this.state });
  };

  render() {
    const {
      username, email, password, errorMessage,
    } = this.state;
    const {
      loading, successMessage,
    } = this.props;

    return (
      <div className="signup-form">
        <GridContainer
          textAlign="center"
          className="register"
          style={{ height: '100%' }}
          verticalAlign="middle"
          divided="vertically"
          columns={2}
        >

          <GridItem>
            <Header
              as="h2"
              textAlign="center"
            >
              Sign Up.
            </Header>
            <Header
              as="h4"
              textAlign="center"
            >
              Enter the email address associated with your account,
              and weâ€™ll send a magic link to your inbox.
            </Header>
            {successMessage
              ? (
                <Message
                  success
                  header="Your user registration was successful"
                  content={successMessage}
                />
              ) : ''}

            <CustomForm
              className="register-form"
              loading={loading}
              size="large"
              errorMessage={errorMessage}
              handleSubmit={this.handleSubmit}
              buttonName="register"
              inputFields={[
                {
                  fluid: true,
                  name: 'username',
                  id: 'username',
                  placeholder: 'username',
                  type: 'text',
                  value: username,
                  onChange: this.handleInputChange,
                  error: !isEmpty(errorMessage ? errorMessage.username : ''),
                },
                {
                  fluid: true,
                  name: 'email',
                  placeholder: 'E-mail address',
                  type: 'email',
                  value: email,
                  onChange: this.handleInputChange,
                  error: !isEmpty(errorMessage ? errorMessage.email : ''),
                },
                {
                  fluid: true,
                  name: 'password',
                  placeholder: 'Password',
                  type: 'password',
                  value: password,
                  onChange: this.handleInputChange,
                  error: !isEmpty(errorMessage ? errorMessage.password : ''),
                },
              ]}
            />
            <Message warning>
              <Icon name="help" />
              Already have an account?&nbsp;
              <a href="/login">login</a>
              &nbsp;instead.
            </Message>
          </GridItem>
        </GridContainer>
      </div>
    );
  }
}

RegisterPage.defautProps = {
  loading: false,
};

RegisterPage.propTypes = {
  // eslint-disable-next-line max-len
  // eslint-disable-next-line
  // react/forbid-prop-types,react/require-default-props,react/no-unused-prop-types
  errorMessage: PropTypes.object,
  // eslint-disable-next-line react/require-default-props
  successMessage: PropTypes.string,
  // eslint-disable-next-line react/require-default-props
  loading: PropTypes.bool,
  signUpAction: PropTypes.func.isRequired,
};

export const mapStateToProps = ({ auth }) => ({
  errorMessage: auth.errorMessage,
  loading: auth.loading,
  successMessage: auth.successMessage,
});

export default connect(
  mapStateToProps,
  { signUpAction },
)(RegisterPage);
