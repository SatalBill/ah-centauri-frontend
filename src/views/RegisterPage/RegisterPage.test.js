import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import expect from 'expect';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { RegisterPage } from './RegisterPage';

// Snapshot for Register React Component
describe('>>>R E G I S T E R --- Snapshot', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RegisterPage signUpAction={() => jest.fn()}/>);
  });

  it('+++capturing Snapshot of Home', () => {
    const renderedValue = renderer.create(<RegisterPage signUpAction={() => jest.fn()}/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('render form with inputs and buttons', () => {
    expect(wrapper.find('.signup-form')).toBeDefined();
  });
});
