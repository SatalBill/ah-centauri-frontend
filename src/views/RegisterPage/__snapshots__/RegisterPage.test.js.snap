// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>R E G I S T E R --- Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="signup-form"
>
  <div
    className="ui vertically divided center aligned middle aligned grid"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="two column row"
    >
      <div
        className="column"
      >
        <h2
          className="ui center aligned header"
        >
          Register for an account
        </h2>
        {"username":"","email":"","password":"","errorMessage":{}}
        <form
          className="ui large form"
          id="registerForm"
          onSubmit={[Function]}
        >
          <div
            className="ui stacked segment"
          >
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
              >
                <input
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  className="user icon"
                  onClick={[Function]}
                />
              </div>
            </div>
            
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
              >
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="E-mail address"
                  type="email"
                  value=""
                />
                <i
                  aria-hidden="true"
                  className="user icon"
                  onClick={[Function]}
                />
              </div>
            </div>
            
            <div
              className="field"
            >
              <div
                className="ui fluid left icon input"
              >
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <i
                  aria-hidden="true"
                  className="lock icon"
                  onClick={[Function]}
                />
              </div>
            </div>
            
            <button
              className="ui large fluid button"
              id="registerButton"
              onClick={[Function]}
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
        <div
          className="ui warning message"
        >
          <i
            aria-hidden="true"
            className="help icon"
            onClick={[Function]}
          />
          Already signed up?  login  instead.
        </div>
      </div>
    </div>
  </div>
</div>
`;
